src/relation_engine_server/server.py

```diff
@@ -67,23 +67,26 @@ def run_query():
     # Don't allow the user to set the special 'ws_ids' field
     json_body['ws_ids'] = []
     auth_token = auth.get_auth_header()
-    is_adhoc_query = 'query' in json_body
-    # Authorize for RE_ADMIN before fetching any workspace IDs
-    if is_adhoc_query:
-        auth.require_auth_token(roles=['RE_ADMIN'])
```

Comment:
Moved this into the adhoc block because I thought that was a bit more readable.

Reply:
I had it up top in a separate block because I wanted it to be checked up-front before we fetched any workspace IDs.. I vaguely remember doing this based on responses I was seeing while testing it. Not sure if it's really worth having it separate/up-front.. I agree it's more readable